arrows/mvg/applets/init_cameras_landmarks.cxx

```diff
@@ -0,0 +1,513 @@
+      {
+        LOG_WARN(main_logger, "Configuration deemed not valid.");
+      }
+      config = nullptr;
+      return WRITE;
+    }
+    else if(!valid_config)
+    {
+      LOG_ERROR(main_logger, "Configuration not valid.");
+      config = nullptr;
+      return FAIL;
+    }
+
+    return SUCCESS;
+  }
+
+  void register_algorithms()
+  {
+    // register the algorithm implementations
+    std::string rel_plugin_path = kwiver::vital::get_executable_path() +
+    "/../lib/kwiver/modules";
+    kwiver::vital::plugin_manager::instance().add_search_path(rel_plugin_path);
+    kwiver::vital::plugin_manager::instance().load_all_plugins();
+  }
+
+  void initialize()
+  {
+    // Create algorithm from configuration
+    initialize_cameras_landmarks::set_nested_algo_configuration( "initializer",
+                                                                 config,
+                                                                 algorithm );
+  }
+
+  void clear_ptrs()
+  {
+    camera_map_ptr = nullptr;
+    landmark_map_ptr = nullptr;
+    feature_track_set_ptr = nullptr;
+    sfm_constraint_ptr = nullptr;
+  }
+
+  void load_tracks( )
+  {
+    kwiver::vital::path_t in_tracks_path =
+      config->get_value<kwiver::vital::path_t>("input_tracks_file");
+    feature_track_set_ptr =
+      kwiver::vital::read_feature_track_file(in_tracks_path);
+  }
+
+  void load_sfm_constraint( )
+  {
+    if(config == nullptr)
+    {
+      return;
+    }
+
+    sfm_constraint_ptr = std::make_shared<sfm_constraints>();
+    if(config->has_value("video_source") &&
+       config->get_value<std::string>("video_source") != "")
+    {
+      video_input_sptr video_reader;
+      video_input::set_nested_algo_configuration("video_reader", config,
+                                                 video_reader);
+      video_input::get_nested_algo_configuration("video_reader", config,
+                                                 video_reader);
+
+      video_reader->open( config->get_value<std::string>("video_source") );
+      if (video_reader->get_implementation_capabilities()
+        .has_capability(video_input::HAS_METADATA))
+      {
+        kwiver::vital::timestamp currentTimestamp;
+        kwiver::vital::metadata_map::map_metadata_t map_t;
+        while (video_reader->next_frame(currentTimestamp))
+        {
+          auto frame = currentTimestamp.get_frame();
+          auto mdVec = video_reader->frame_metadata();
+
+          if (mdVec.size() > 0)
+          {
+            map_t.emplace(frame, mdVec);
+          }
+        }
+        kwiver::vital::metadata_map_sptr metadata_map =
+        std::make_shared<kwiver::vital::simple_metadata_map>(map_t);
+        sfm_constraint_ptr->set_metadata(metadata_map);
+      }
+    }
+  }
+
+  bool write_cameras()
+  {
+    std::string output_cameras_director =
+    config->get_value<std::string>("output_cameras_director");
+    std::string camera_prefix = config->get_value<std::string>("camera_prefix");
```

Comment:
See also other comment about camera prefix.  This was the old way of naming output files.  We now use the video metadata to set the name of these output files.